Source/DafnyLanguageServer/DafnyCodeActions.cs

```diff
@@ -33,22 +33,105 @@ public class DafnyCodeActions {
+  }
+
   public static List<DafnyCodeAction> RemoveAction(string title, Diagnostic diagnostic, Range range) {
     var edit = new DafnyCodeActionEdit[] { new DafnyCodeActionEdit(range, "") };
     var action = new InstantDafnyCodeAction(title, new List<Diagnostic> { diagnostic }, edit);
     return new List<DafnyCodeAction> { action };
   }
 
+  public static void AddRemoveAction(ErrorID errorID, String title) {
+    codeActionMap.Add(errorID, (Diagnostic diagnostic, Range range) => RemoveAction(title, diagnostic, range));
+  }
+
+  public static void AddReplaceAction(ErrorID errorID, String title, String newContent) {
+    codeActionMap.Add(errorID, (Diagnostic diagnostic, Range range) => ReplacementAction(title, diagnostic, range, newContent));
+  }
+
+  public static void AddReplacementActions(ErrorID errorID, Tuple<String, String>[] actions) {
+    codeActionMap.Add(errorID, (Diagnostic diagnostic, Range range) => ReplacementActions(diagnostic, range, actions));
+  }
+
   private static bool initialized = false;
   public static void init() {
     if (initialized) {
       return;
     }
     initialized = true;
-    codeActionMap.Add(ErrorID.p_bad_const_initialize_op, (Diagnostic diagnostic, Range range) => ReplacementAction("replace = with :=", diagnostic, range, ":="));
-    codeActionMap.Add(ErrorID.p_abstract_not_allowed, (Diagnostic diagnostic, Range range) => RemoveAction("remove 'abstract'", diagnostic, range));
-    codeActionMap.Add(ErrorID.p_no_leading_underscore, (Diagnostic diagnostic, Range range) => RemoveAction("remove underscore", diagnostic, range));
+    AddRemoveAction(ErrorID.p_duplicate_modifier, "remove duplicate modifier'");
+    AddRemoveAction(ErrorID.p_abstract_not_allowed, "remove 'abstract'");
+    AddRemoveAction(ErrorID.p_no_ghost_for_by_method, "remove 'ghost'");
+    AddRemoveAction(ErrorID.p_ghost_forbidden_default, "remove 'ghost'");
+    AddRemoveAction(ErrorID.p_ghost_forbidden, "remove 'ghost'");
+    AddRemoveAction(ErrorID.p_no_static, "remove 'static'");
+    AddRemoveAction(ErrorID.p_deprecated_attribute, "remove attribute");
+    AddReplaceAction(ErrorID.p_literal_string_required, "replace with empty string", "\"\"");
+    AddRemoveAction(ErrorID.p_no_leading_underscore, "remove underscore");
+    // p_bitvector_too_large -- no code action
+    // p_array_dimension_too_large -- no code action
+    // p_superfluous_exoirt -- TODO - need a code action
+    // p_bad_module_decl -- no code action
+    AddRemoveAction(ErrorID.p_extraneous_comma_in_export, "remove comma");
+    AddReplaceAction(ErrorID.p_top_level_field, "replace 'var' by 'const'", "const"); // also remove entire declaration?
+    // p_bad_datatype_refinement -- what code action
+    AddReplaceAction(ErrorID.p_no_mutable_fields_in_value_types, "replace 'var' by 'const'", "const");
+    AddReplaceAction(ErrorID.p_bad_const_initialize_op, "replace = with :=", ":=");
+    // p_const_is_missing_type_or_init -- no code action
```

Comment:
Could you perhaps add a code action named "show me an example" that inserts `: int := 2` ?